packages/bpk-component-infinite-scroll/src/DataSource.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+class DataSource {
```

Comment:
Can we make this type generic over the type of the elements in the `DataSource`? i.e.

```javascript
class DataSource<T=any> {
  fetchItems(index: number, nElements: number): Promise<Array<T>> {
    throw new Error('Not implemented');
  }
}
```

Reply:
what's the difference between using `T` and `any`?

Reply:
If we support a generic argument that defaults to `any` it will behave the same as you've written by default. However if the consumer explicitly specialises the type with e.g. `DataSource<number` Flow can help catch more errors in the code because it knows the type. For example on `ArrayDataSource<number>` the specialised type of the `Promise` will prevent incorrect usage of the return values

Reply:
👍 changed